converters/fromZigbee.js

```diff
@@ -1440,23 +1440,43 @@ const converters = {
             return {presence: true};
         },
     },
-    moes_105z_dimmer: {
+    moes_105_dimmer: {
         cluster: 'manuSpecificTuya',
         type: ['commandGetData', 'commandSetDataResponse'],
         convert: (model, msg, publish, options, meta) => {
+            const multiEndpoint = model.meta && model.meta.multiEndpoint;
             const dp = msg.data.dp;
             const value = tuya.getDataValue(msg.data.datatype, msg.data.data);
 
-            meta.logger.debug(`from moes_105z_dimmer, msg.data.dp=[${dp}], msg.data.datatype=[${msg.data.datatype}], value=[${value}]`);
+            meta.logger.debug(`from moes_105_dimmer, msg.data.dp=[${dp}], msg.data.datatype=[${msg.data.datatype}], value=[${value}]`);
 
-            switch (dp) {
-            case tuya.dataPoints.state:
-                return {state: value ? 'ON': 'OFF'};
-            case tuya.dataPoints.moes105zDimmerLevel:
-                return {brightness: mapNumberRange(value, 0, 1000, 0, 254)};
-            default:
-                meta.logger.debug(`zigbee-herdsman-converters:moes_105z_dimmer:` +
-                    `NOT RECOGNIZED DP #${dp} with data ${JSON.stringify(msg.data)}`);
+            const state = value ? 'ON': 'OFF';
+            const brightness = mapNumberRange(value, 0, 1000, 0, 254);
+
+            if (multiEndpoint) {
+                switch (dp) {
+                case tuya.dataPoints.moes105DimmerState1:
+                    return {state_l1: state};
```

Comment:
if you change this to `return {[multiEndpoint ? 'state' : 'state_l1']: state};` and also do this for `moes105DimmerLevel1` the `if/else` is not needed anymore

Reply:
ok, fixed it (needed to swap 'state_l1' and 'state' from your comment), thanks!